### PR TITLE
gameflow: enhance save crystals and game mode selections

### LINKED ISSUE #1102: Save crystal mode and Unfinished Business
Ran into a couple issues:

- Unfinished Business doesn't have save crystals, so it doesn't really make sense for the config option to prevent saving when launching the game with `-gold`.

- This led me to try disabling `enable_save_crystals` and only enabling `force_enable_save_crystals` in `TR1X_gameflow.json5`. Problem with this method is that `force_enable_save_crystals` actually overwrites your config's `enable_save_crystals` value, so if you play base TR and then later launch UB, save crystal mode will be enabled again.

I suppose another possibility would be to import the save crystal locations from b122251's PSX port of UB.

### PR DESCRIPTION
**Pull request by rr-**
Resolves #1102

#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

- changed `force_enable_save_crystals` to `force_save_crystals` to let custom level authors force-disable save crystals
- changed `force_disable_game_modes` to `force_game_modes` to let custom level authors force-enable game modes
- fixed inability to save in Unfinished Business in crystals mode as UB doesn't have crystals

### CODE REVIEW

File: bin/cfg/TR1X_gameflow.json5 [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "main_menu_picture": "data/titleh.png",
     "savegame_fmt_legacy": "saveati.%d",
     "savegame_fmt_bson": "save_tr1_%02d.dat",
-    "force_disable_game_modes": false,
-    "force_enable_save_crystals": false,
+    "force_game_modes": null,
+    "force_save_crystals": false,
```
Comment (lahm86) [inline bin/cfg/TR1X_gameflow.json5]: I think this should be `null` - with `false` there is no way to play with save crystals without manually changing this line (so the config option is ignored).

Comment (rr-) [reply bin/cfg/TR1X_gameflow.json5]: That's correct, my mistake with copypasting.

**Review by lahm86 — APPROVED** [review]

**Review by walkawayy — APPROVED** [review]
Looks good

**Comment by walkawayy** [comment]
Only comment for changelog is, if I understand this correctly, the custom level author can force enable or also force disable crystals or modes. The changelog makes it seem like only it only allows force enable (which would mean the user could then choose in the config).

**Comment by rr-** [comment]
Thanks – I'll change it

**Review by walkawayy — APPROVED** [review]

**Pull request merged**